internal/check/valid_owner.go

```diff
@@ -192,7 +192,52 @@ func (v *ValidOwner) validateTeam(ctx context.Context, name string) *validateErr
 	}
 
 	if !teamExists() {
-		return newValidateError("Team %q does not exist in organization %q or has no permissions associated with the repository.", team, org)
+		return newValidateError("Team %q does not exist in organization %q.", team, org)
+	}
+
+	// repo contains the permissions for the team slug given
+	repo, _, err := v.ghClient.Teams.IsTeamRepoBySlug(ctx, v.orgName, team, org, v.orgRepoName)
```

Comment:
Thanks for adding that functionality, it was also requested some time ago (https://github.com/mszostok/codeowners-validator/issues/21)

Unfortunately, this approach increases the number of external call to GitHub, probably in the near feature it will be good to switch to GraphQL which will remove problem with over fetched data, and we will be able to merge the `ListTeams` and `IsTeamRepoBySlug` into single query,  see: https://github.com/mszostok/codeowners-validator/issues/21#issuecomment-716044156

but this is for future, for now, it lgtm 👍

Reply:
That is quite true. There are also potentially issues with GraphQL, which we've experienced with another project we contribute to, the GitHub Terraform provider – some calls because extremely, extremely slow in larger orgs (branch protection) and it annihilated the performance compared to REST. I doubt it'd be an issue here, just sharing our experience.

Reply:
I'll add a todo.

Reply:
thx for the heads-up 👍